static/panes/compiler.ts

```diff
@@ -2557,22 +2562,22 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
         // The executorButton does not need to be changed here, because you can create however
         // many executors as you want.
 
-        this.optButton.toggle(!!(this.compiler as any).supportsOptOutput);
-        this.ppButton.toggle(!!(this.compiler as any).supportsPpView);
-        this.astButton.toggle(!!(this.compiler as any).supportsAstView);
-        this.irButton.toggle(!!(this.compiler as any).supportsIrView);
-        this.llvmOptPipelineButton.toggle(!!(this.compiler as any).supportsLLVMOptPipelineView);
-        this.deviceButton.toggle(!!(this.compiler as any).supportsDeviceAsmView);
-        this.rustMirButton.toggle(!!(this.compiler as any).supportsRustMirView);
-        this.rustMacroExpButton.toggle(!!(this.compiler as any).supportsRustMacroExpView);
-        this.rustHirButton.toggle(!!(this.compiler as any).supportsRustHirView);
-        this.haskellCoreButton.toggle(!!(this.compiler as any).supportsHaskellCoreView);
-        this.haskellStgButton.toggle(!!(this.compiler as any).supportsHaskellStgView);
-        this.haskellCmmButton.toggle(!!(this.compiler as any).supportsHaskellCmmView);
-        this.cfgButton.toggle(!!(this.compiler as any).supportsCfg);
-        this.gccDumpButton.toggle(!!(this.compiler as any).supportsGccDump);
-        this.gnatDebugTreeButton.toggle(!!(this.compiler as any).supportsGnatDebugViews);
-        this.gnatDebugButton.toggle(!!(this.compiler as any).supportsGnatDebugViews);
+        this.optButton.toggle(!!this.compiler.supportsOptOutput);
```

Comment:
I think we can remove the `!!`'s here, I wonder if there's a way to get eslint to point this out

Reply:
oh that's cool if so. The `!!` are no doubt FUD from my earliest js code. A lot of my original JS learnings come from jsbeeb, which used a pidgin set of js to try and trick out the js JIT a lot...so...many...bad...habits

Reply:
Ahha I see :)

Reply:
Ok, I tried removing these but had to re-introduce. It turns out they're critical, otherwise gcc thinks it can spit out LLVM IR 😄 
![image](https://user-images.githubusercontent.com/51220084/212140501-22b8d6f2-3ca2-4e64-aacf-2a753fa8ab14.png)
I guess JQuery.toggle doesn't like undefined values. I went ahead and updated the `CompilerInfo.supports*` entries to indicate they can be undefined.